cirq-core/cirq/sim/state_vector.py

```diff
@@ -314,7 +314,8 @@ def measure_state_vector(
     out.shape = initial_shape
     out /= np.sqrt(probs[result])
 
-    return measurement_bits, out
+    # We mutate and return out, so mypy cannot identify that the out cannot be None.
```

Comment:
Incidentally, I think the type annotation on `out` should be `Optional[np.ndarray]`. Might as well fix it while you're in the neighborhood.

Reply:
Good catch.  Fixed.